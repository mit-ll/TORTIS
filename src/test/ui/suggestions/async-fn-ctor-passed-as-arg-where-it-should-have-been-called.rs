// edition:2018
use std::future::Future;

async fn foo() {}

fn bar(f: impl Future<Output=()>) {}

fn main() {
    bar(foo); //~ERROR E0277
}
