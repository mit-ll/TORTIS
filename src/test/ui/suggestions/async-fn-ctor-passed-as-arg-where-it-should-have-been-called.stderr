error[E0277]: the trait bound `fn() -> impl std::future::Future {foo}: std::future::Future` is not satisfied
  --> $DIR/async-fn-ctor-passed-as-arg-where-it-should-have-been-called.rs:9:9
   |
LL | fn bar(f: impl Future<Output=()>) {}
   |    ---         ----------------- required by this bound in `bar`
...
LL |     bar(foo);
   |         ^^^
   |         |
   |         the trait `std::future::Future` is not implemented for `fn() -> impl std::future::Future {foo}`
   |         help: use parentheses to call the function: `foo()`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
